src/sentry/models/release.py

```diff
@@ -116,24 +116,30 @@ def get_or_create(cls, project, version, date_added):
             if not release:
                 release = cls.objects.filter(
                     organization_id=project.organization_id,
-                    version=version
+                    version='%s-%s' % (project.slug, version),
```

Comment:
We should just do this on the initial query and change it to:

```python
cls.objects.filter(
  version__in=[version, '%s-%s' % (project.slug, version)],
)
```

Then have a tiny bit of logic to pick the right one. Like, if 1 row returned, just go with it, etc.